File: src/main/java/JavaChallenge/MissingNumber.java
package JavaChallenge;

public class MissingNumber {

    public static void main(String[] args) {

        int[] arr1 = {3, 0, 1}; // 2
        int[] arr2 = {0, 1}; // 2
        int[] arr3 = {9, 6, 4, 2, 3, 5, 7, 0, 1}; // 8

        System.out.println(missingNumber(arr1));
        System.out.println(missingNumber(arr2));
        System.out.println(missingNumber(arr3));

    }

    public static int missingNumber(int[] numbs) {

        int sum1 = 0, sum2 = 0, lengthOfNumbs = numbs.length;
        for (int i = 0; i <= lengthOfNumbs; i++) sum1 += i;
        for (int numb : numbs) sum2 += numb;
        System.gc(); // with this, 100.00 % Beats for Space C. Without GC Time C will be 100.00%
        return sum1 - sum2;

    }
}

File: src/main/kotlin/LeetCodeSolutions/MissingNumber.kt
package LeetCodeSolutions

object MissingNumber {
    @JvmStatic
    fun main(args: Array<String>) {

        println(missingNumber(
            intArrayOf(3, 0, 1) // 2
        ))
        println(missingNumber(
            intArrayOf(0, 1) // 2
        ))
        println(missingNumber(
            intArrayOf(9, 6, 4, 2, 3, 5, 7, 0, 1) // 8
        ))

    }

    private fun missingNumber(numbs: IntArray): Int {
        var sum1 = 0
        var sum2 = 0
        val lengthOfNumbs = numbs.size
        for (i in 0..lengthOfNumbs) sum1 += i
        for (numb in numbs) sum2 += numb
        System.gc() // with this, 100.00 % Beats for Space C. Without GC Time C will be 100.00%
        return sum1 - sum2
    }
}